app/assets/javascripts/live_search.js

```diff
@@ -134,50 +160,63 @@
   }
 
   LiveSearch.prototype.formChange = function formChange (e) {
-    var pageUpdated
     if (this.isNewState()) {
-      this.getTaxonomyFacet().update()
+      this.getAndUpdateTaxonomyFacet()
       this.saveState()
       this.updateOrder()
       this.updateLinks()
       this.updateTitle()
-      pageUpdated = this.updateResults()
-      pageUpdated.done(
-        function () {
-          var newPath = window.location.pathname + '?' + $.param(this.state)
-          window.history.pushState(this.state, '', newPath)
-          this.trackingInit()
-          this.setRelevantResultCustomDimension()
-          this.trackPageView()
-        }.bind(this)
-      )
+      this.updateResults()
+    }
+  }
+
+  LiveSearch.prototype.serializeState = function (state) {
+    var params = []
+    if (Array.isArray(state)) {
+      for (var i = 0; i < state.length; i++) {
+        params.push(state[i].name + '=' + state[i].value)
```

Comment:
I think it's safe to assume that the `name` will already include `[]` if that's appropriate (e.g. if it's a checkbox or radio). Otherwise we might end up with things like `vessel_type[][]=merchant-vessel`.

On the encoding question, the state gets encoded in a single line in `updateResults` (currently on line 370) which feels more efficient to do it all in one gone that potentially call encodeURIComponent hundreds of times.

Reply:
I put some logging into the code and this function is called with arrays and objects in what looks to be broadly similar numbers when the jasmine tests are run. A quick test of the code running in a browser showed only calls with arrays passed, but I haven't comprehensively checked all the possible searches.